app/models/device.rb

```diff
@@ -43,4 +43,10 @@ def change_privilege_for(dev, new_privilege)
     record.privilege = new_privilege
     record.save
   end
+
+  def device_checkin_hash
+    hash = self.as_json
+    hash[:last_checkin] = self.checkins.last.get_data if self.checkins.exists?
```

Comment:
@t-harps self is the first thing that's called in the call stack
Try just taking it out and seeing if it still works :)